src/traces/image/style.js

```diff
@@ -13,6 +13,7 @@ var d3 = require('d3');
 module.exports = function style(gd) {
     d3.select(gd).selectAll('.im image')
         .style('opacity', function(d) {
-            return d.trace.opacity;
+            if(d && d.hidden) return 0;
+            return d[0].trace.opacity;
```

Comment:
Or in one line:
```js
return (d && d.hidden) ? 0 : d[0].trace.opacity;
```

Reply:
Also wondering if there is a bug fix involved here?
before: 
```js
return d.trace.opacity; 
```
now: 
```js
return d[0].trace.opacity; 
```

Reply:
This line is not necessary anymore. :hocho: in fe1d2083c4ae61d7c9ee85561fb4654f4cde4643

Reply:
> Also wondering if there is a bug fix involved here?
> before:
> 
> ```js
> return d.trace.opacity; 
> ```
> 
> now:
> 
> ```js
> return d[0].trace.opacity; 
> ```

@antoinerg Could you elaborate on this?

Reply:
In a prior commit, I changed the d3 selection's data from an object (`.data(cd)`) to an array of objects (`.data([cd])`). There's no bugfix, simply a leftover. I could go back to `.data(cd)` instead of `.data[cd]` and leave `hover.js` intact.